deploy/local/LOCAL_DEPLOYMENT_TESTING.md

```diff
@@ -0,0 +1,197 @@
+- tags the opencodelists image and deploys to the existing Dokku instance
+- cleans up old Docker images to free up space
+
+
+## Working with Coding System Databases
+
+By default the script does not copy across any coding system databases.
+
+If you need them to test viewing a codelist the best approach is as follows:
+
+1. Attempt to view the codelist in the browser. (This may need to be a published or under review codelist.) - you'll get a 500 error.
+
+If you're feeling brave you can run this script to automatically view the logs, find the missing database, and copy it across. If you're more cautious, move to step 2.
+
+```bash
+latest_db=$(docker exec -t opencodelists-local-deployment bash -c "docker logs opencodelists.web.1" | grep -oE 'coding_systems/[^ ]+\.sqlite3' | tail -n1)
+if [[ -z "$latest_db" ]]; then
+    echo "No missing database found in logs."
+else
+    echo "Latest missing database: $latest_db"
+    echo "Attempting to copy the database file..."
+    docker cp "$latest_db" opencodelists-local-deployment:/var/lib/dokku/data/storage/opencodelists/$latest_db
+    echo "Database file copied - try to load the codelist again."
+fi
+```
+
+2. Check the logs for the web service container to see which database is missing by executing the following command:
+
+```bash
+docker exec -t opencodelists-local-deployment bash -c "docker logs opencodelists.web.1"
+```
+
+You will see an error similar to:
+
+```bash
+django.db.utils.OperationalError: no such table: snomedct_description
+
+If this is development then you may be missing the following sqlite database:
+
+- coding_systems/snomedct/snomedct_3920_20241127.sqlite3
+```
+
+3. Copy the missing database file. So for the example above you would run:
+
+```bash
+MISSING_DB=coding_systems/snomedct/snomedct_3920_20241127.sqlite3
+docker cp $MISSING_DB opencodelists-local-deployment:/var/lib/dokku/data/storage/opencodelists/$MISSING_DB
+```
+
+## Troubleshooting
+
+The first step is to trigger a rebuild with
+
+```bash
+FORCE_REBUILD=1 just local-deploy
+```
+
+Other things that may be useful:
+
+```bash
+# Check container logs
+docker logs opencodelists-local-deployment
+
+# Check inner container logs for the opencodelists app
+docker exec -t opencodelists-local-deployment bash -c "docker logs opencodelists.web.1"
+
+# Check inner container logs for the dokku container
+docker exec -t opencodelists-local-deployment docker logs dokku
+```
+
+## TODO
+
+- These scripts don't do any of the CI stuff from the main.yml and deploy.yml files. A lot of that is linting and testing which can be achieved with `just` commands. But there are some things like the smoke-test which may fail. Currently the smoke-test will fail as soon as you push your branch, so it will be caught before deployment. Therefore it's not essential to replicate that here, but it might be useful to add in the future.
```

Comment:
thought: I think this is fine for now. The benefit is that we can least try out the deployment.